api/client/samples/similar_regions_Frechet/sr.py

```diff
@@ -0,0 +1,560 @@
+        If 'no', take only cached data (subjcted to OK_TO_PROCEED_REGION_FRACTION threshold), nothing read from Gro API.
+        If 'update', read cached data, then try to read from API whatever is missing.
+        :param region_info_reload: If True, reload region info from Gro, otherwise try to use cached (still reload if region not found).
+        :param t_int_per_year: into how many intervals to split calendar year when generating distributions from raw data.
+        Default is 52 (roughly weekly). High values (such as 365 - daily) lead to greater storage requirements and not recommended
+        :param metric_instance: A {property:weight} dictionary stating specific properties and their weights to use for this run.
+        If not provided, all properties from metric_properties are used with equal weights
+        :param drop_mode: If 'any_missing', drop all regions which do nor have full set of valid datapoints
+        (i.e. t_int_per_year data points for each property). If 'fully_missing', drop region only if there is less than 2 valid datapoints
+        """
+        self.data = None
+        self.t_int_per_year = t_int_per_year
+        self.t_int_days = 365 / self.t_int_per_year # number of days in single t interval (can be float)
+        self.to_retry = []
+        self.available_regions = []
+        self.available_properties = []
+        self._logger = get_default_logger()
+        self._logger.setLevel(logging.INFO)
+        update=update_mode
+
+        # extract needed parameters from inputs
+        self.metric_properties = metric_properties
+        self.needed_properties = sorted(list(set(metric_properties.keys())))
+        if metric_instance is not None:
+            unresolved_items = [item for item in metric_instance if item not in self.metric_properties]
+            assert not unresolved_items, "Found items {} in metric instance not present in metric description".format(unresolved_items)
+            self.metric_instance = metric_instance
+        else:
+            self.metric_instance = dict(zip(self.metric_properties.keys(),[1.0]*len(self.metric_properties)))
+            
+        if data_dir:
+            self.data_dir = data_dir
+        elif os.path.isdir(CACHE_PATH) and os.access(CACHE_PATH, os.W_OK):
+            self.data_dir = CACHE_PATH
+        else:
+            # no access to CACHE_PATH - likely doesn't exist => attempt to create directory
+            try:
+                os.mkdir(CACHE_PATH)
+                self.data_dir = CACHE_PATH
+            except:
+                # last resort
+                self.data_dir = tempfile.gettempdir()
+        
+        self._logger.info("*********Creating similar region object with *******\n\tregion {}\n\tupdate_mode {}\n\tdata_dir {}\n\tregion_info_reload {}\n\tt_int_per_year {}\n".format(search_region,update_mode,self.data_dir,region_info_reload,t_int_per_year))
+
+        self.client = BatchClient(API_HOST, ACCESS_TOKEN)
```

Comment:
you can use directly GroClient, BatchClient is exactly the same with last Gro api release

Reply:
If you say so (don't think it was the case when I started this). But I guess it then does not hurt to keep BatchClient (i.e. as it is now) either unless we plan to deprecate it which I assume we don't. Keeping as is should avoid the need for new testing.

Reply:
done